src/generic/Matrix.jl

```diff
@@ -6042,18 +6047,20 @@ end
 ################################################################################
 
 @doc Markdown.doc"""
-    identity_matrix(R::Ring, n::Int) -> MatElem
+    identity_matrix(R::Ring, n::Int)
 
 Return the $n \times n$ identity matrix over $R$.
 """
 identity_matrix(R::Ring, n::Int) = diagonal_matrix(one(R), n)
 
 @doc Markdown.doc"""
-    identity_matrix(M::MatElem{T}) where T <: RingElement
+    identity_matrix(M::MatElem{T}, [n::Int]) where T <: RingElement
```

Comment:
Look, I don't know if this is just some kind of GitHub glitch, but what I clearly see here is a change to a docstring that does not match the function it is documenting. There is no [n::Int] argument in the function, so I don't see why the docstring should be changed to that.

But perhaps GitHub is playing some kind of silly game here. I feel like attaching a screenshot so you can see what I mean.

Reply:
It's not a GitHub glitch. Although this kind of situtation is not ideal, it's sometimes prefered by some authors to attach a docstring which describes two distinct methods to only one of the methods in order to limit repetetion. One current example of this is the docstring for `similar` in Julia `Base` which is attached only to the method `similar(a::AbstractArray{T}) where {T}`, although the docstring describes more methods, with optional `element_type` and optional `dims`.

A slightly cleaner way is to attach the same docstring to both methods (there is a syntax for that, basically add after the docstring a list a `,`-separated signatures). Here having two distinct docstrings would also be fine I think, as the duplication would be low.

Reply:
We literally spent 3 months of HiWi time making sure all docstrings match the functions they are documenting. I'm not at all happy about reversing course on that. I think we should find another way.

Reply:
An alternative here is also to attach the docstring to the function, instead of the methods, e.g. 
```julia
"""
    identitty_matrix(M, [n::Int]) 

desciption
""" 
function identity_matrix end
```

Reply:
Here I basically copied the appearance from the `diagonal_matrix` docs a couple of lines down. Could that approach wrong as well?

Reply:
I think so, yes. The reason is that quite clearly someone will come along in a month, long after it's forgotten why it's like that, and remove this, thinking that it's just a mistake. In my opinion it should be perfectly obvious that this is not the right way to solve this problem.

Of course I completely understand that you modeled it on what had already been done. I'm objecting to every instance of it, not just yours. Of course someone else may overrule me. I'm not the one making the decisions.